python/pyspark/worker.py

```diff
@@ -3167,7 +3170,28 @@ def func(_, it):
     return func, None, ser, ser
 
 
+def write_profile(outfile):
+    import yappi
+
+    while True:
+        stats = []
+        for thread in yappi.get_thread_stats():
+            data = list(yappi.get_func_stats(ctx_id=thread.id))
+            stats.extend([{str(k): str(v) for k, v in d.items()} for d in data])
+        pickled = pickle.dumps(stats)
```

Comment:
While pickle? Would JSON maybe make more sense so we can interpert more easily in say the Spark UI in the future?